example/ssh2_echo.c

```diff
@@ -256,7 +256,10 @@ int main(int argc, char *argv[])
                 continue;
 
             if (fds[0].revents & LIBSSH2_POLLFD_POLLIN) {
-                int n = libssh2_channel_read(channel, buffer, sizeof(buffer));
+                int n;
+                libssh2_session_set_blocking(session, 1);
```

Comment:
This uses blocking mode for reads and writes that would block, while `ssh2_echo` is claiming to `tell libssh2 we want it all done non-blocking`.

If only reads should happen on the same session while read returns EAGAIN, shouldn't the read/write loops be changed to do that instead?

I'm not sure that is a limitation though, have not come across it and have tried interleaving reads with writes in non-blocking mode. There is at least [one other example script](https://www.libssh2.org/examples/tcpip-forward.html) that does the same and works fine, as does `ssh2_echo` for me.